ArchiSteamFarm/Steam/Integration/ArchiHandler.cs

```diff
@@ -151,6 +154,110 @@ public sealed class ArchiHandler : ClientMsgHandler {
 		return response.Result == EResult.OK ? response.GetDeserializedResponse<CCredentials_LastCredentialChangeTime_Response>() : null;
 	}
 
+	[PublicAPI]
+	public async IAsyncEnumerable<Asset> GetMyInventoryAsync(uint appID = Asset.SteamAppID, ulong contextID = Asset.SteamCommunityContextID, bool tradableOnly = false, bool marketableOnly = false) {
+		ArgumentOutOfRangeException.ThrowIfZero(appID);
+		ArgumentOutOfRangeException.ThrowIfZero(contextID);
+
+		if (Client.SteamID == null) {
+			throw new InvalidOperationException(nameof(Client.SteamID));
+		}
+
+		SteamID steamID = Client.SteamID;
+		ushort itemsCountPerRequest = 50_000;
+
+		// We need to store asset IDs to make sure we won't get duplicate items
+		HashSet<ulong>? assetIDs = null;
+		ulong startAssetID = 0;
+
+		while (true) {
+			// Steam still does not give us any descriptions, throw error to the user
+			if (itemsCountPerRequest < 5000) {
+				throw new NotSupportedException(nameof(itemsCountPerRequest));
+			}
+
+			ulong currentStartAssetID = startAssetID;
+
+			ushort currentItemsCountPerRequest = itemsCountPerRequest;
+
+			CEcon_GetInventoryItemsWithDescriptions_Request request = new() {
+				appid = appID,
+				contextid = contextID,
+				filters = new CEcon_GetInventoryItemsWithDescriptions_Request.FilterOptions {
+					tradable_only = tradableOnly,
+					marketable_only = marketableOnly
+				},
+				get_descriptions = true,
+				steamid = steamID.ConvertToUInt64(),
+				start_assetid = currentStartAssetID,
+				count = currentItemsCountPerRequest
+			};
+
+			SteamUnifiedMessages.ServiceMethodResponse genericResponse = await UnifiedEconService
+				.SendMessage(x => x.GetInventoryItemsWithDescriptions(request))
+				.ToLongRunningTask()
+				.ConfigureAwait(false);
+
+			CEcon_GetInventoryItemsWithDescriptions_Response response = genericResponse.GetDeserializedResponse<CEcon_GetInventoryItemsWithDescriptions_Response>();
+
+			if ((response.total_inventory_count == 0) || (response.assets.Count == 0)) {
+				// Empty inventory
+				yield break;
+			}
+
+			if (response.descriptions.Count == 0) {
+				// Looks like rate-limiting, reducing count per request helps
+				itemsCountPerRequest /= 2;
```

Comment:
One thing I'd like to see:

- Extract `itemsCountPerRequest` as optional argument in the function call. I'd appreciate if you could do some research and find the minimum value that works *always*, if you're unable to find one or don't want, set `5000` as default one since this is what Steam uses.

The argument can be of type `ushort`, we won't limit max, we can keep existing logic, I just want default call to work in a way we won't need to trigger this.